packages/core/src/types/data/data.ts

```diff
@@ -0,0 +1,36 @@
+import { DataType } from '@tensorflow/tfjs-node-gpu';
+import { Statistic } from '../other';
+
+export interface DataDescriptor {
+  type?: DataType;
+  shape?: number[];
+  featureNames?: string[];
+}
+
+export interface MetaData {
+  feature?: DataDescriptor;
+  label?: DataDescriptor;
+  labelMap?: {[key: string]: number};
+}
+
+export interface Sample {
+  data: any;
+  label: any;
+}
+
+export interface DataLoader {
+  len: () => Promise<number>;
+  getItem: (id: number) => Promise<Sample>;
+}
+
+export interface UniDataset {
+  metaData?: MetaData;
+  dataStatistics: Statistic[];
+  validationResult: {
```

Comment:
Is this for model validation?

Reply:
this is for dataset validation in case some plugins want to do some checks for data in data access step